File: MyPortal.Core/Entities/Role.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using MyPortal.Common.Enums;
using MyPortal.Core.Interfaces;

namespace MyPortal.Core.Entities
{
    [Table("Roles")]
    public class Role : Entity, ISystemEntity
    {
        public string? Description { get; set; }
        public bool IsSystem { get; set; }

        public UserType UserType { get; set; }

        public bool IsDefault { get; set; }

        // Identity
        
        [StringLength(256)]
        public string? Name { get; set; }
        
        [StringLength(256)]
        public string? NormalizedName { get; set; }
        
        public string? ConcurrencyStamp { get; set; }
    }
}

File: MyPortal.Frontend/myportal-spa/src/app/shared/types/role.ts
import { UserType } from '../../core/types/user-type';

export interface RoleSummaryResponse {
  id: string;
  description: string | null;
  isSystem: boolean;
  isDefault: boolean;
  userType: UserType;
  name: string | null;
}

export interface RoleDetailsResponse {
  id: string;
  description: string | null;
  isSystem: boolean;
  isDefault: boolean;
  userType: UserType;
  name: string | null;
  permissionIds: string[];
}

export interface RoleUpsertRequest {
  name: string;
  description: string | null;
  userType: UserType;
  permissionIds: string[];
}

export interface PermissionResponse {
  id: string;
  name: string;
  friendlyName: string;
  area: string;
  userType: UserType;
}
